docs/configs-spec2006-tests/run_spec.py

```diff
@@ -0,0 +1,250 @@
+                      num_cpus = 1, # run the benchmark in a single thread
+                      no_kvm = False,
+                      TimingCPUModel = detailed_cpu_model)
+    
+    # For workitems to work correctly
+    # This will cause the simulator to exit simulation when the first work
+    # item is reached and when the first work item is finished.
+    system.work_begin_exit_count = 1
+    system.work_end_exit_count = 1
+
+    # set up the root SimObject and start the simulation
+    root = Root(full_system = True, system = system)
+
+    if system.getHostParallel():
+        # Required for running kvm on multiple host cores.
+        # Uses gem5's parallel event queue feature
+        # Note: The simulator is quite picky about this number!
+        root.sim_quantum = int(1e9) # 1 ms
+
+    return root, system
+
+
+def boot_linux():
+    '''
+    Output 1: False if errors occur, True otherwise
+    Output 2: exit cause
+    '''
+    exit_event = m5.simulate()
+    return exit_event.getCause() == "m5_exit instruction encountered", exit_event.getCause()
+
+def run_spec_benchmark():
+    '''
+    Output 1: False if errors occur, True otherwise
+    Output 2: exit cause
+    '''
+    exit_event = m5.simulate()
+    return exit_event.getCause() == "m5_exit instruction encountered", exit_event.getCause()
+
+def copy_spec_logs():
+    '''
+    Output 1: False if errors occur, True otherwise
+    Output 2: exit cause
+    '''
+    exit_event = m5.simulate()
+    return exit_event.getCause() == "m5_exit instruction encountered", exit_event.getCause()
+
+if __name__ == "__m5_main__":
+    args = parse_arguments()
+
+    cpu_name = args.cpu
+    benchmark_name = getBenchmarkName(args.benchmark)
+    benchmark_size = args.size
+    linux_kernel_path = args.kernel
+    disk_image_path = args.disk
+    no_copy_logs = args.no_copy_logs
+    no_listeners = args.no_listeners
+
+    output_dir = os.path.join(m5.options.outdir, "speclogs")
+
+    # Get the DetailedCPU class from its name
+    detailed_cpu = getDetailedCPUModel(cpu_name)
+    if detailed_cpu == None:
+        print("'{}' is not define in the config script.".format(cpu_name))
+        print("Change getDeatiledCPUModel from run_spec.py to add more CPU Model.")
+        exit(1)
+
+    if not benchmark_size in ["ref", "train", "test"]:
+        print("Benchmark size must be one of the following: ref, train, test")
+        exit(1)
+
+    root, system = create_system(linux_kernel_path, disk_image_path, detailed_cpu)
+
+    # Create and pass a script to the simulated system to run the reuired
+    # benchmark
+    system.readfile = writeBenchScript(m5.options.outdir, benchmark_name,
+                                       benchmark_size, output_dir)
+
+    # needed for long running jobs
+    if no_listeners:
+        m5.disableAllListeners()
+
+    # instantiate all of the objects we've created above
+    m5.instantiate()
+
+    # booting linux
+    print("Booting Linux")
```

Comment:
Maybe this should be in the `boot_linux` function if you want to keep the specific function names.

Reply:
Yeah, I think it's a better way to do this.